File: traittable/fib/fib.js
function fib(n) {
  if(n < 2) return 1;
  return fib(n-2)+fib(n-1);
}
function Fib(n) {
  this.n = n;
}
Fib.prototype.fib = function() {
  if (this.n < 2) return 1;
  return new Fib(this.n-2).fib() + new Fib(this.n-1).fib();
}

var start = new Date().getTime();
console.log(fib(40));
console.log(new Date().getTime()-start+"ms");

var start = new Date().getTime();
console.log(new Fib(40).fib());
console.log(new Date().getTime()-start+"ms");

File: traittable/fib/fib.rs
#![crate_id(name = "fib")]

extern crate std;
extern crate time;

fn fib(n : int) -> int {
  if n < 2 {
    return 1;
  } else {
    return fib(n - 1) + fib(n - 2);
  }
}

trait Fib {
	fn fib(&self) -> int;
}

impl Fib for int {
	fn fib(&self) -> int {
		if (self+0 < 2) {
			1
		} else {
			(self-2).fib() + (self-1).fib()
		}
	}
}

fn tim() -> i64 {
	let t = time::get_time();
	(t.sec * 1000) + ((t.nsec / 1000000) as i64)
}
fn main() {
  let start = tim();
  println!("{}", fib(40));
  println!("{}ms", tim()-start);

  let start = tim();
  println!("{}", 40.fib());
  println!("{}ms", tim()-start);
}

